### DISCUSSION TITLE
August Newsletter

### DISCUSSION THREAD

**Discussion by briancl2** [briancl2/CustomerNewsletter#5] [Customer Newsletter]
This is a personally curated newsletter for just my customers. The purpose is to pull together and share current resources, announcements, and events. I welcome your feedback, and please let me know if you would like to add or remove anyone from this list.

### Optimize and Measure Copilot Usage
Last month I shared a podcast with a panel of engineering leaders discussing AI adoption. One element they stressed was the importance of running developer surveys to start measuring something as it relates to the impact of AI tools in software development. Every source of data related to software development and development productivity has its strengths and weaknesses, but one thing that is consistent across all well-run DevEx organizations is their commitment to surveying developers directly.

For this month, I’d like to share another [podcast from the same host](https://getdx.com/podcast/dora-research-google/) that dives much deeper into running developer surveys including a discussion on DORA metrics, survey methodology, and survey design. Don’t forget to check out [GitHub’s Copilot Learning Pathway](https://resources.github.com/learn/pathways/copilot/essentials/measuring-the-impact-of-github-copilot/) which includes [example developer surveys](https://downloads.ctfassets.net/wfutmusr1t3h/66acuCKYqXme0aukY8Rn3x/a8c682946b0176db5860544ad85fffe7/2024-04-23-GitHub-CCI-LP-Copilot-Impact-Survey-NT-V003.pdf).

Additional Copilot Metrics Updates:
* [GitHub Action](https://github.com/marketplace/actions/copilot-user-report) to retrieve Copilot seat usage for an enterprise or organization
* [Public repo](https://github.com/austenstone/copilot-license-cleanup) to automatically remove inactive Copilot licenses 
* Metrics API beta: [Enterprise Team metrics now available](https://github.blog/changelog/2024-08-08-enterprise-team-metrics-now-available-on-the-copilot-metrics-api/). This is a long awaited feature to slice Copilot metrics based on teams of 5 or more Copilot users for fine grained reporting.

### Releases and Roadmap
* Complete GitHub Roadmap update [recording from Aug 8](https://www.youtube.com/watch?v=QyGwvU2F0Fs)
* Copilot Extensions is now [accepting waitlist registrations](https://github.blog/changelog/2024-08-13-sign-up-for-the-github-copilot-extensions-waitlist/) from all users
* Copilot Workspace is still in limited beta. I stumbled upon the most [comprehensive preview on YouTube](https://www.youtube.com/watch?v=TONH_vqieYc) from a user that recorded his first attempt at using Workspace to tackle a real-world issue.
* Actions doesn’t get a lot of coverage, but we’ve shipped many enhancements including our [ARM64 hosted runners](https://github.blog/news-insights/product-news/arm64-on-github-actions-powering-faster-more-efficient-build-systems/) which can provide significant cost savings. Customer quote: “We switched to the GitHub ARM runners from a custom, self-managed setup on AWS Graviton instances. Switching to GitHub runners has saved us over 75% on our monthly fees and removed all the management overhead.”
* Found Means Fixed: [Copilot Autofix for GHAS is now GA](https://github.blog/news-insights/product-news/secure-code-more-than-three-times-faster-with-copilot-autofix/)

<img width="915" alt="Screenshot 2024-08-29 at 3 52 48 PM" src="https://github.com/user-attachments/assets/b63b81fa-5343-4b3c-83cb-545b5fb034ca">

### Industry News
* AWS announced it will no longer accept new customers for AWS CodeCommit. See more details on how that could impact you and learn about migration options on [our blog](https://github.blog/enterprise-software/collaboration/github-enterprise-the-best-migration-path-from-aws-codecommit/)
* The Pragmatic Engineer published [Part 2](https://newsletter.pragmaticengineer.com/p/ai-tooling-part-2) of their AI Tooling survey which includes a wealth of direct user quotes that cover the spectrum of Good, Bad, and Ugly:
  * “GitHub Copilot has been a game changer for day-to-day productivity in our company.”
  * 25 year vet says “Still learning how to use it.” It takes time and experimentation to figure things out.
  * GitHub Copilot is the “only” tool that works. Several people say Copilot is the only tool they’re not entirely dissatisfied with, after testing several.
* [Gartner names GitHub clear leader in first MQ for AI Code Assistants](https://github.blog/news-insights/company-news/github-named-a-leader-in-the-gartner-first-ever-magic-quadrant-for-ai-code-assistants/)
* [Why culture is so important in the age of AI](https://colinsalmcorner.com/why-culture-is-so-important/) by GitHub’s own Colin Dembovsky

### Recent Recordings
* Aug 16 - [CodeCompass: Open Source AI for Personalized GitHub Discovery](https://www.youtube.com/watch?v=8Thz9na2Zbk) 
* Aug 28 - [AMA: GitHub Copilot - Coding with an AI pair programmer](https://www.youtube.com/watch?v=5drqCXv16js)
* Brian’s personally curated Copilot Training & Tips [Youtube playlist](https://www.youtube.com/playlist?list=PLCiDM8_DsPQ1WJ5Ss3e0Lsw8EaijUL_6D), updated monthly. This contains a list of videos from the community that cover onboarding, new user training, and advanced user cases.

### Upcoming Webinars
* Sept 11 - [GitHub Copilot Prompt Engineering 101](https://developer.microsoft.com/en-us/reactor/events/23470/)
* Sept 23 - [Get your on-prem apps AI-ready using GitHub Copilot and Azure Migrate Code Assessment](https://developer.microsoft.com/en-us/reactor/events/23588/)
* Oct 1 - [End-to-End AI application deployments with GitHub Copilot](https://developer.microsoft.com/en-us/reactor/events/23594/)
* Oct 3 - [Extending GitHub Copilot to chat with your Postgres Database](https://developer.microsoft.com/en-us/reactor/events/23595/)
* Oct 8 - [10x Developer with GitHub Copilot](https://developer.microsoft.com/en-us/reactor/events/23597/)
* Oct 15 - [Create a website using GitHub Copilot and Codespaces](https://developer.microsoft.com/en-us/reactor/events/23579/)
* Oct 22 - [Automate your repository using GitHub Actions](https://developer.microsoft.com/en-us/reactor/events/23581/)
* Oct 22 - [Learn to Code using GitHub Copilot and Codespaces](https://developer.microsoft.com/en-us/reactor/events/23630/)

### Upcoming Events
* NYC - September 12: NY Yankees Suite with Endor Labs
* Chicago - September 19: Chicago Cubs Rooftop
* San Francisco - Oct 29-30: [Discounted Tickets](https://githubuniverse.com/#tickets) available through September 3

**Comment by alwell-kevin** [comment]
Excellent letter, as always 🎉! Thanks for sharing.

**Comment by RaduButa3** [comment]
❤️ this @briancl2 !!
